homeassistant/components/netatmo/binary_sensor.py

```diff
@@ -35,130 +34,61 @@
-                if (
-                    config[CONF_CAMERAS] != []
-                    and camera_name not in config[CONF_CAMERAS]
-                ):
-                    continue
-            for variable in welcome_sensors:
-                add_entities(
-                    [
-                        NetatmoBinarySensor(
-                            data,
-                            camera_name,
-                            module_name,
-                            home,
-                            timeout,
-                            camera_type,
-                            variable,
-                        )
-                    ],
-                    True,
-                )
-        if camera_type == "NOC":
-            if CONF_CAMERAS in config:
-                if (
-                    config[CONF_CAMERAS] != []
-                    and camera_name not in config[CONF_CAMERAS]
-                ):
-                    continue
-            for variable in presence_sensors:
-                add_entities(
-                    [
-                        NetatmoBinarySensor(
-                            data,
-                            camera_name,
-                            module_name,
-                            home,
-                            timeout,
-                            camera_type,
-                            variable,
-                        )
-                    ],
-                    True,
-                )
+    def get_devices():
+        """Retrieve Netatmo devices."""
+        devices = []
+
+        try:
+            data = CameraData(hass, auth)
+        except NoDevice:
+            _LOGGER.debug("No camera devices to add")
+
+        for camera_id in data.get_all_camera_ids():
+            camera_type = data.get_camera_type(camera_id=camera_id)
+            home_id = data.get_camera_home_id(camera_id=camera_id)
 
-        for module_name in data.get_module_names(camera_name):
-            for variable in tag_sensors:
-                camera_type = None
-                add_entities(
-                    [
-                        NetatmoBinarySensor(
-                            data,
-                            camera_name,
-                            module_name,
-                            home,
-                            timeout,
-                            camera_type,
-                            variable,
-                        )
-                    ],
-                    True,
+            for sensor_name in SENSOR_TYPES[camera_type]:
+                devices.append(
+                    NetatmoBinarySensor(data, camera_id, home_id, sensor_name)
                 )
 
+        return devices
+
+    async_add_entities(await hass.async_add_executor_job(get_devices), True)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the access to Netatmo binary sensor."""
+    return
+
 
 class NetatmoBinarySensor(BinarySensorDevice):
```

Comment:
Do you have enough data to add a `unique_id` property to this entity?

Reply:
Maybe something like home id - module id - sensor type ?

Reply:
Yes, added that now.